File: apps/api/schemas/metrics.py
from __future__ import annotations

from pydantic import BaseModel


class StatusCount(BaseModel):
    status: str
    count: int


class PriorityCount(BaseModel):
    priority: str
    count: int


class TopError(BaseModel):
    exception_type: str
    count: int


class MetricsResponse(BaseModel):
    total_incidents: int
    total_analyzed: int
    by_status: list[StatusCount]
    by_priority: list[PriorityCount]
    top_errors: list[TopError]

File: apps/dashboard/src/types/metrics.ts
export interface StatusCount {
  status: string
  count: number
}

export interface PriorityCount {
  priority: string
  count: number
}

export interface TopError {
  exception_type: string
  count: number
}

export interface MetricsResponse {
  total_incidents: number
  total_analyzed: number
  by_status: StatusCount[]
  by_priority: PriorityCount[]
  top_errors: TopError[]
}
